src/cmlibs/utils/zinc/mesh.py

```diff
@@ -279,3 +282,65 @@ def undefine_field(field):
             mesh = fm.findMeshByDimension(i)
             mesh_group = mesh
             _undefine_field_on_elements(field, mesh_group)
+
+
+def calculate_jacobian(coordinates):
+    """
+    Calculate the Jacobian for the input coordinate field.
+    If the coordinate field is not suitable for calculating
+    the Jacobian return None.
+    :param coordinates: Geometric coordinate field.
+    :return: Jacobian field.
+    """
+    jacobian = None
+    if coordinates is not None and coordinates.isValid() and coordinates.getNumberOfComponents() == 3:
+        fm = coordinates.getFieldmodule()
+        jacobian = fm.createFieldDotProduct(
+            fm.createFieldCrossProduct(
+                fm.createFieldDerivative(coordinates, 1),
+                fm.createFieldDerivative(coordinates, 2)),
+            fm.createFieldDerivative(coordinates, 3))
+        jacobian.setName("Jacobian")
+
+    return jacobian
+
+
+def report_on_lowest_value(real_field, group_field=None):
```

Comment:
Would a better name be get_scalar_field_minimum_in_mesh()? It's not actually reporting here.

Note: It would be free to return the maximum too as it's already calculated in field range, and it could even handle multiple components, but this is where the FieldRange is designed for the future: it already has the ability to store per-component element:xi locations for minima and maxima, but there isn't a public method to grow the range rather that overwrite it. I'm not motivated to modify Zinc to do this at this time, so maybe it's best this utility function doing scalar field minimum and in future think about returning the FieldRange to handle both minima and maximum and multiple components.

Reply:
Second argument should be a mesh, which can also be a mesh_group. Best to leave it to the caller to find mesh_3d or mesh_group as they see fit.
(I reserve the ability to have multiple full meshes in a region at some point, so best to get caller to supply.)

Reply:
group_field is actually a mesh_group, so was improperly named anyway.